diff --git a/wazo_agid/handlers/queue.py b/wazo_agid/handlers/queue.py
index 9550a74..0fd51ee 100644
--- a/wazo_agid/handlers/queue.py
+++ b/wazo_agid/handlers/queue.py
@@ -16,6 +16,8 @@ USERSHAREDLINES_CONTEXT_RE = re.compile(r'^Local/(.+)@usersharedlines-[a-f0-9]+;
 
 
 class AnswerHandler(handler.Handler):
+    destination_agent_id = None
+
     def execute(self):
         try:
             callee = self.get_user()
@@ -34,6 +36,7 @@ class AnswerHandler(handler.Handler):
         if result:
             agent_id = result.group(1)
             search_params = {'agent_id': int(agent_id)}
+            self.destination_agent_id = agent_id
         elif usersharedlines_match := USERSHAREDLINES_CONTEXT_RE.match(channel_name):
             user_uuid = usersharedlines_match.group(1)
             if user_uuid:
@@ -53,10 +56,12 @@ class AnswerHandler(handler.Handler):
         external = call_origin == 'extern'
         logger.debug('Queue call origin is %s', call_origin)
         internal = not external
+        queue_recording = self._agi.get_variable('WAZO_QUEUE_RECORDING')
         should_record = any(
             [
                 internal and callee.call_record_incoming_internal_enabled,
                 external and callee.call_record_incoming_external_enabled,
+                self.destination_agent_id and queue_recording,
             ]
         )
         logger.debug(
@@ -66,6 +71,16 @@ class AnswerHandler(handler.Handler):
         )
 
         if not should_record:
+            self._agi.verbose(
+                (
+                    'Call recording is not enabled for call of type "{}" '
+                    'for callee "{}"(uuid={})'
+                ).format(
+                    "external" if external else "internal",
+                    " ".join([callee.firstname, callee.lastname]),
+                    callee.uuid,
+                )
+            )
             return
 
         calld = self._agi.config['calld']['client']
@@ -73,6 +88,12 @@ class AnswerHandler(handler.Handler):
         tenant_uuid = self._agi.get_variable('WAZO_TENANT_UUID')
         self._agi.set_variable('WAZO_RECORD_QUEUE_CALLEE', '1')
         try:
+            self._agi.verbose(
+                'Initiating call recording for callee "{}"(uuid={})'.format(
+                    " ".join([callee.firstname, callee.lastname]),
+                    callee.uuid,
+                )
+            )
             calld.calls.start_record(channel_id, tenant_uuid=tenant_uuid)
         except Exception as e:
             logger.error('Error during enabling call recording: %s', e)
diff --git a/wazo_agid/modules/call_recording.py b/wazo_agid/modules/call_recording.py
index 4dbb309..16df094 100644
--- a/wazo_agid/modules/call_recording.py
+++ b/wazo_agid/modules/call_recording.py
@@ -25,6 +25,14 @@ CALL_RECORDING_FILENAME_TEMPLATE = (
     '/var/lib/wazo/sounds/tenants/{tenant_uuid}/monitor/{recording_uuid}.wav'
 )
 
+CALL_RECORDING_FILENAME_TEMPLATE_A = (
+    '/var/lib/wazo/sounds/tenants/{tenant_uuid}/monitor/{recording_uuid}-a.wav'
+)
+
+CALL_RECORDING_FILENAME_TEMPLATE_B = (
+    '/var/lib/wazo/sounds/tenants/{tenant_uuid}/monitor/{recording_uuid}-b.wav'
+)
+
 
 def call_recording(agi: FastAGI, cursor: DictCursor, args: list[str]) -> None:
     calld = agi.config['calld']['client']
@@ -39,6 +47,51 @@ def call_recording(agi: FastAGI, cursor: DictCursor, args: list[str]) -> None:
         _enable_call_recording(agi, calld, channel_id, tenant_uuid)
 
 
+BINAURAL_MIXMONITOR_COMMAND = (
+    '/usr/bin/sox -M -v 1 {filename_a} -v 1 {filename_b} {filename} '
+    '&& rm {filename_a} {filename_b}'
+)
+BINAURAL_MIXMONITOR_OPTIONS = 'r({filename_a})t({filename_b})'
+
+
+def _get_mixmonitor_args(agi: FastAGI, *args: list[Any]) -> tuple[str, str, str]:
+    tenant_uuid = agi.get_variable(dv.TENANT_UUID)
+    recording_uuid = str(uuid.uuid4())
+    filename = CALL_RECORDING_FILENAME_TEMPLATE.format(
+        tenant_uuid=tenant_uuid,
+        recording_uuid=recording_uuid,
+    )
+    filename_a = CALL_RECORDING_FILENAME_TEMPLATE_A.format(
+        tenant_uuid=tenant_uuid,
+        recording_uuid=recording_uuid,
+    )
+    filename_b = CALL_RECORDING_FILENAME_TEMPLATE_B.format(
+        tenant_uuid=tenant_uuid,
+        recording_uuid=recording_uuid,
+    )
+    mix_monitor_options = agi.get_variable('WAZO_MIXMONITOR_OPTIONS')
+
+    binaural_options = BINAURAL_MIXMONITOR_OPTIONS.format(
+        filename_a=filename_a,
+        filename_b=filename_b,
+    )
+    mixmonitor_command = BINAURAL_MIXMONITOR_COMMAND.format(
+        filename=filename,
+        filename_a=filename_a,
+        filename_b=filename_b,
+    )
+    new_mixmonitor_options = f'{mix_monitor_options}{binaural_options}'
+    return filename, new_mixmonitor_options, mixmonitor_command
+
+
+def setup_binaural_mixmonitor(
+    agi: FastAGI, cursor: DictCursor, args: list[str]
+) -> None:
+    filename, mixmonitor_options, mixmonitor_command = _get_mixmonitor_args(agi)
+    agi.set_variable('__WAZO_MIXMONITOR_OPTIONS', mixmonitor_options)
+    agi.set_variable('__WAZO_MIXMONITOR_COMMAND', mixmonitor_command)
+
+
 def record_caller(agi: FastAGI, cursor: DictCursor, args: list[str]) -> None:
     is_being_recorded = agi.get_variable('WAZO_CALL_RECORD_ACTIVE') == '1'
     if is_being_recorded:
@@ -97,25 +150,16 @@ def start_mix_monitor(agi, cursor, args):
 
 
 def _start_mix_monitor(agi):
-    tenant_uuid = agi.get_variable(dv.TENANT_UUID)
-    recording_uuid = str(uuid.uuid4())
-    filename = CALL_RECORDING_FILENAME_TEMPLATE.format(
-        tenant_uuid=tenant_uuid,
-        recording_uuid=recording_uuid,
+    filename, mixmonitor_options, mixmonitor_command = _get_mixmonitor_args(agi)
+    agi.appexec(
+        'MixMonitor',
+        f'{filename},{mixmonitor_options},{mixmonitor_command}',
     )
-    mix_monitor_options: str = agi.get_variable('WAZO_MIXMONITOR_OPTIONS')
-    if 'p(' in mix_monitor_options:
-        p_position = mix_monitor_options.index('p(')
-        ending_paren_position = mix_monitor_options.index(')')
-        pre_p = mix_monitor_options[:p_position]
-        post_p = mix_monitor_options[ending_paren_position + 1 :]
-        mix_monitor_options = pre_p + post_p
-
-    agi.appexec('MixMonitor', f'{filename},{mix_monitor_options}')
-    agi.set_variable('WAZO_RECORDING_UUID', recording_uuid)
+
     agi.set_variable('WAZO_CALL_RECORD_ACTIVE', '1')
 
 
 agid.register(call_recording)
 agid.register(record_caller)
 agid.register(start_mix_monitor)
+agid.register(setup_binaural_mixmonitor)
