diff --git a/dialplan/asterisk/extensions_lib_queue.conf b/dialplan/asterisk/extensions_lib_queue.conf
index b24c77e2..97c7fa0a 100644
--- a/dialplan/asterisk/extensions_lib_queue.conf
+++ b/dialplan/asterisk/extensions_lib_queue.conf
@@ -34,6 +34,8 @@ same  =   n,AGI(agi://${WAZO_AGID_IP}/check_diversion)
 same  =   n,GotoIf(${XIVO_DIVERT_EVENT}?DIVERT,1)
 same  =   n,AGI(agi://${WAZO_AGID_IP}/queue_skill_rule_set)
 same  =   n,GoSub(wazo-subroutine,s,1(${XIVO_QUEUEPREPROCESS_SUBROUTINE}))
+; setup Queue pre-dial hook for binaural recording
+same  =   n,GoSub(setup-binaural-record-pre,s,1)
 same  =   n,Set(XIVO_QUEUETIMEOUT=${IF($["${ARG2_TIMEOUT}" != ""]?${ARG2_TIMEOUT}:${XIVO_QUEUETIMEOUT})})
 
 ; announcing holdtime
@@ -102,3 +104,18 @@ exten = s,1,AGI(agi://${WAZO_AGID_IP}/queue_answered_call)
 same = n,GotoIf(${WAZO_QUEUE_SUBROUTINE}?:nosub)
 same = n,GoSub(${WAZO_QUEUE_SUBROUTINE},s,1)
 same = n(nosub),Return()
+
+[setup-binaural-record-pre]
+exten = s,1,NoOp(preparing Queue invocation for binaural recording)
+same = n,Set(WAZO_QUEUEOPTIONS=${WAZO_QUEUEOPTIONS}b(setup-binaural-record^s^1))
+same = n,Return()
+
+[setup-binaural-record]
+exten = s,1,NoOp(Setting up binaural recording options)
+same = n,AGI(agi://${WAZO_AGID_IP}/setup_binaural_mixmonitor)
+same = n,Return()
+
+[enable-queue-recording]
+exten = s,1,NoOp(Enabling recording for queue ${WAZO_QUEUENAME})
+same = n,Set(__WAZO_QUEUE_RECORDING=1)
+same = n,Return()
